ninewinecfg/main.c

```diff
@@ -337,36 +388,48 @@ static void nine_set(BOOL status, BOOL NoOtherArch)
         else if (isWoW64())
             Call64bitNineWineCfg(status);
     }
-
-    /* Delete unused DllRedirects key */
-    common_del_registry_key(reg_path_dll_redirects, reg_key_d3d9);
-
-    /* enable native dll */
-    if (!status)
-    {
-        if (!common_del_registry_key(reg_path_dll_overrides, reg_key_d3d9))
-            ERR("Failed to delete 'HKCU\\%s\\%s'\n'", reg_path_dll_overrides, reg_key_d3d9);
-    }
     else
```

Comment:
Why?
It's been a while and it's all rather ugly, but we want that code to run on boths archs,no?

Reply:
No. Nine create only 1 reg in `HKCU\\Software\\Wine\\DllOverrides\\`, even if it was called 2 times for different archs. So you don't need to remove it twice.

Reply:
Hm, wow64 redirect doesn't apply tp HKCU/Software, but doesn't that skip that code if you manually invoke `ninewinecfg -n`?

Reply:
When you call `ninewinecfg -n`, you got `NoOtherArch = TRUE`, `if (!NoOtherArch)` return false and you go to `else` and run same original behavior/code:
```
        /* Delete unused DllRedirects key */
        common_del_registry_key(reg_path_dll_redirects, reg_key_d3d9);

        /* enable native dll */
        if (!status)
        {
            if (!common_del_registry_key(reg_path_dll_overrides, reg_key_d3d9))
                ERR("Failed to delete 'HKCU\\%s\\%s'\n'", reg_path_dll_overrides, reg_key_d3d9);
        }
        else
        {
            if (!common_set_registry_string(reg_path_dll_overrides, reg_key_d3d9, reg_value_override))
                ERR("Failed to write 'HKCU\\%s\\%s'\n", reg_path_dll_overrides, reg_key_d3d9);
        }
```

Reply:
Right. But I'd still tend to keep that hunk as-is. It's been so long, but I remember vaguely that there was some reason.
Maybe if you build nine only for one arch, because now you rely on the "other" arch to wipe those keys?

Reply:
I still don't fully understand what was reason for it, but I returned the original code, because it was just tweak)